scripts/check-prompt-refs.sh

```
@@ -0,0 +1,161 @@
+# Usage:
+#   scripts/check-prompt-refs.sh              # default instruction-file scope
+#   scripts/check-prompt-refs.sh FILE...      # explicit files (for testing;
+#                                             # /dev/stdin reads piped text and
+#                                             # exercises check (a))
+#
+# No `set -e`: greps that legitimately find nothing exit 1 mid-pipeline, so
+# correctness is by explicit violation accumulation + a single final exit 1,
+# not by errexit.
+set -uo pipefail
+
+cd "$(dirname "${BASH_SOURCE[0]}")/.."
```

Shellcheck (0.11.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)